brave/src/test/java/com/linecorp/armeria/it/brave/BraveIntegrationTest.java

```diff
@@ -448,19 +450,27 @@ void testSpanInThreadPoolHasSameTraceId() throws Exception {
     }
 
     @Test
-    void testServerTimesOut() throws Exception {
-        assertThatThrownBy(() -> timeoutClient.hello("name"))
-                .isInstanceOf(TTransportException.class)
-                .hasCauseInstanceOf(InvalidResponseHeadersException.class);
-        final MutableSpan[] spans = spanHandler.take(2);
-
-        final MutableSpan serverSpan = findSpan(spans, "service/timeout");
-        final MutableSpan clientSpan = findSpan(spans, "client/timeout");
-
-        // Server timed out meaning it did still send a timeout response to the client and we have all
-        // annotations.
-        assertThat(serverSpan.annotations()).hasSize(2);
-        assertThat(clientSpan.annotations()).hasSize(2);
+    void testServerTimesOut() {
+        try (ClientFactory cf = ClientFactory.builder().build()) {
+            final Iface timeoutClient =
+                    ThriftClients.builder(server.httpUri())
+                                 .path("/timeout")
+                                 .factory(cf)
+                                 .decorator(BraveClient.newDecorator(newTracing("client/timeout")))
+                                 .build(Iface.class);
+            assertThatThrownBy(() -> timeoutClient.hello("name"))
+                    .isInstanceOf(TTransportException.class)
+                    .hasCauseInstanceOf(InvalidResponseHeadersException.class);
+            final MutableSpan[] spans = spanHandler.take(2);
+
+            final MutableSpan serverSpan = findSpan(spans, "service/timeout");
+            final MutableSpan clientSpan = findSpan(spans, "client/timeout");
+
+            // Server timed out meaning it did still send a timeout response to the client and we have all
+            // annotations.
+            assertThat(serverSpan.annotations()).hasSize(2);
+            assertThat(clientSpan.annotations()).hasSize(8);
```

Comment:
The number of annotations depends on whether the `ClientFactory` has existing connections, so this test has been refactored